src/types/UiaTextRangeBase.cpp

```diff
@@ -1000,22 +1000,49 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         auto inclusiveEnd = _end;
         bufferSize.DecrementInBounds(inclusiveEnd, true);
 
+        // reserve size in accordance to extracted text
         const auto textRects = buffer.GetTextRects(_start, inclusiveEnd, _blockRange, true);
-        const auto bufferData = buffer.GetText(true,
-                                               false,
-                                               textRects);
+        if (maxLength > 0)
+        {
+            textData.reserve(gsl::narrow<size_t>(maxLength));
+        }
+        else if (textRects.size() == 1)
+        {
+            const auto rect = textRects.front();
+            textData.reserve(base::ClampedNumeric<size_t>(rect.right) - rect.left + 1);
+        }
+        else
+        {
+            const auto firstRect = textRects.front();
+            const auto firstWidth = firstRect.right - firstRect.left + 1;
+
+            const auto lastRect = textRects.back();
+            const auto lastWidth = lastRect.right - lastRect.left + 1;
+
+            const auto otherWidth = base::ClampedNumeric<size_t>(textRects.size()) - 2 * bufferSize.Width();
+            textData.reserve(firstWidth + otherWidth + lastWidth);
+        }
 
-        const size_t textDataSize = bufferData.text.size() * bufferSize.Width();
-        textData.reserve(textDataSize);
-        for (const auto& text : bufferData.text)
+        // actually add the text
+        for (const auto& rect : textRects)
         {
+            const auto bufferData = buffer.GetText(true,
+                                                   false,
+                                                   { rect });
+            const auto& text = bufferData.text.front();

```

Comment:
I'm worried that moving from "request multiple rects simultaneously" to "request one rect at a time" will break TextBuffer's understanding of line wrapping. In general, does this change cause other trouble?

I'm sorry I pushed you into overcomplicating this PR. That's on me. I think Leonard was right -- we should keep what we had and fix it up in whatever ways we can. The cost of extra copying is minimal and won't really be a problem.

Reply:
reverted to older implementation (with a minor reordering)